parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,347 @@
+				// If SubstitutionScheduleDay is not empty
+				if (substitutionScheduleDay.getSubstitutions() != null) {
+					substitutionSchedule.addDay(substitutionScheduleDay);
+				}
+				for (int k = 0; k < 7; k++) {
+					final LocalDate date = LocalDate.now().plusDays(k);
+					if ((date.isAfter(startingDate) || date.isEqual(startingDate)) &&
+						(date.isBefore(endingDate) || date.isEqual(endingDate))) {
+						substitutionScheduleDay = new SubstitutionScheduleDay();
+						substitutionScheduleDay.setDate(date);
+						substitutionScheduleDay.addSubstitution(substitution);
+						substitutionSchedule.addDay(substitutionScheduleDay);
+					}
+				}
+				break;
+			}
+			// If starting date of change does not equal date of SubstitutionScheduleDay
+			if (!startingDate.isEqual(currentDate)) {
+				// If SubstitutionScheduleDay is not empty
+				if (substitutionScheduleDay.getSubstitutions() != null) {
+					substitutionSchedule.addDay(substitutionScheduleDay);
+				}
+				substitutionScheduleDay = new SubstitutionScheduleDay();
+				substitutionScheduleDay.setDate(startingDate);
+			}
+			substitutionScheduleDay.addSubstitution(substitution);
+		}
+		substitutionSchedule.addDay(substitutionScheduleDay);
+	}
+	
+	private Substitution getSubstitution(JSONObject change, HashMap<String, String> coursesHashMap, HashMap<String, String> teachersHashMap) throws IOException, JSONException {
+		final Substitution substitution = new Substitution();
+		// Set class
+		final String classId = change.getString("course");
+		if (!classId.equals("0") && !classId.equals(null)) {
+			if (coursesHashMap == null) {
+				throw new IOException("Change references a course but courses are empty.");
+			}
+			final String singleClass = coursesHashMap.get(classId);
+			final HashSet<String> classes = new HashSet<>();
+			classes.add(singleClass);
+			substitution.setClasses(classes);
+		}
+		// Set type
+		String type = "Unknown";
+		switch (change.getString("type")) {
+			case "0":
+				type = "Entfall";
+				break;
+			case "1":
+				type = "Vertretung";
+				break;
+			case "2":
+				type = "Information";
+				break;
+		}
+		substitution.setType(type);
+		// Set color
+		substitution.setColor(colorProvider.getColor(type));
+		// Set covering teacher
+		final String coveringTeacherId = change.getString("coveringTeacher");
+		if (!coveringTeacherId.equals("0")) {
+			if (teachersHashMap == null) {
+				throw new IOException("Change references a covering teacher but teachers are empty.");
+			}
+			substitution.setTeacher(teachersHashMap.get(coveringTeacherId));
+		}
+		// Set teacher
+		final String teacherId = change.getString("teacher");
+		if (!teacherId.equals("0")) {
+			if (teachersHashMap == null) {
+				throw new IOException("Change references a teacher but teachers are empty.");
+			}
+			if (type.equals("Vertretung") || !coveringTeacherId.equals("0")) {
+				substitution.setPreviousTeacher(teachersHashMap.get(teacherId));
+			}
+			else {
```

Comment:
`}` and `else` should be put on the same line in Java (I think this way is more common in C#).